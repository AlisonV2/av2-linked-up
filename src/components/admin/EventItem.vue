<template>
  <div class="card text-center">
    <div class="card-header title">
      {{ event.name }}
    </div>
    <p class="card-text">{{ event.description }}</p>
    <p class="card-text">{{ event.localisation }}</p>
    <div class="card-footer text-muted">
      {{ event.date }}
    </div>
  </div>
</template>

<script>
import * as Sentry from '@sentry/vue';

/**
 * @exports EventItem
 * @type {Component}
 * @vue-prop {object} event - event object
 * @vue-data {string} role - User role
 */
export default {
  name: 'eventItem',
  props: ['event'],
  data() {
    return {
      role: '',
    };
  },
  created() {
    this.$store
      .dispatch('getUserRole')
      .then(() => {
        this.role = this.$store.getters.getUserRole;
      })
      .catch((err) => {
        Sentry.captureException(err);
      });
  },
};
</script>
