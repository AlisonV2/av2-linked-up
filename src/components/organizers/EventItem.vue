<template>
  <div class="card Event-card">
    <div class="card-overlay"></div>
    <div class="card-details fade-in-top text-center">
      <h5 class="card-title" :data-test="event.id">{{ event.name }}</h5>
      <p class="card-text">{{ event.city }} ({{ event.zip }})</p>
    </div>
    <img
      :src="event.thumbnail"
      class="card-img-top"
      :alt="event.id"
    />
    <div class="card-body">
      <h5 class="card-title">{{ event.name }}</h5>
    </div>
  </div>
</template>

<script>
/**
 * @exports EventItem
 * @type {Component}
 * @vue-prop {object} events
 */
export default {
  name: 'EventItem',
  props: ['event'],
};
</script>

<style lang="scss" scoped>
.event-card {
  .card-title {
    font-family: $accent-font;
    color: $primary;
    font-size: 1.2rem;
  }
  .card-text {
    color: $light;
  }
  .card-details {
    position: absolute;
    text-align: center;
    padding-left: 1em;
    padding-right: 1em;
    width: 100%;
    top: 50%;
    left: 50%;
    opacity: 0;
    transform: translate(-50%, -50%);
    transition: all 0.3s ease-in-out 0s;
    .card-title {
      margin-bottom: 2rem;
    }
  }
  .card-overlay {
    background: rgba(0, 0, 0, 0.8);
    position: absolute;
    height: 100%;
    width: 100%;
    left: 0;
    top: 0;
    bottom: 0;
    right: 0;
    opacity: 0;
    transition: all 0.4s ease-in-out 0s;
  }
}

.Event-card:hover {
  .card-overlay {
    opacity: 1;
  }
  .card-details {
    display: block;
    top: 50%;
    left: 50%;
    opacity: 1;
  }
}

.fade-in-top {top: 20%};
</style>
