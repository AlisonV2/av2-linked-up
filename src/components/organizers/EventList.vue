<template>
  <div class="row artists-list">
    <div
      class="col-12 col-md-6 col-lg-4 col-xl-3"
      v-for="event in events"
      :key="event.id"
    >
      <router-link :to="{ name: 'Event', params: { id: event.id } }">
        <EventItem :event="event" />
      </router-link>
    </div>
  </div>
</template>

<script>
import EventItem from '@/components/organizers/EventItem';
import * as Sentry from '@sentry/vue';

/**
 * @exports ArtistList
 * @type {Component}
 * @vue-data {array} events
 * @vue-event created - Dispatch store action
 */
export default {
  name: 'EventList',
  components: {
    EventItem,
  },
  data() {
    return {
      events: [],
    };
  },
  created() {
    this.$store
      .dispatch('getEvents')
      .then(() => {
        this.events = this.$store.getters.getEvents;
      })
      .catch((err) => {
        Sentry.captureException(err);
      });
  },
};
</script>

<style lang="scss">
@include bp-down(md) {
  .artists-list {
    padding-right: 1rem;
    padding-left: 1rem;
  }
}
</style>
