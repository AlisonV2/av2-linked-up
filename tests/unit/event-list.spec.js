import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import { createStore } from 'vuex';
import EventList from '@/components/organizers/EventList';
import EventItem from '@/components/organizers/EventItem';

const actions = {
  getEvents: jest.fn(),
};

const getters = {
  getEvents: jest.fn(),
};

const store = createStore({
  actions,
  getters,
});

const config = {
  data() {
    return { events: [] };
  },
  global: {
    plugins: [store],
    components: {
      'router-link': RouterLinkStub,
      EventItem: EventItem,
    },
  },
};

/**
 * @module EventListTest
 */
describe('EventList.vue', () => {
  it('EventList Snapshot', () => {
    const component = shallowMount(EventList, config);
    expect(component.element).toMatchSnapshot();
  });

  it('getEvents action is called on created', () => {
    const component = shallowMount(EventList, config);
    EventList.created.call(component.vm);
    expect(actions.getEvents).toHaveBeenCalled();
  });

  it('getEvents getter is called on created', () => {
    const component = shallowMount(EventList, config);
    EventList.created.call(component.vm);
    expect(getters.getEvents).toHaveBeenCalled();
  });
});
