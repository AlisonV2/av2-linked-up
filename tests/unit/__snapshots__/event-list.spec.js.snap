// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventList.vue EventList Snapshot 1`] = `
<div
  class="row artists-list"
>
  
  
</div>
`;
